ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/PreExecutingValidatingCommitter.scala

```diff
@@ -102,13 +90,17 @@ class PreExecutingValidatingCommitter[StateValue, ReadSet, WriteSet](
             ),
           )
         } yield {
-          submissionAggregator.finish()
           submissionResult
-        }
+        }).transform {
+          case Failure(_: ConflictDetectedException) =>
+            logger.error("Too many conflicts detected during post-execution. Giving up.")
+            Success(SubmissionResult.Acknowledged) // But it will simply be dropped.
+          case result => result
+        }.map(result => {
+          submissionAggregator.finish()
```

Comment:
Can we confirm that this is the desired behavior?

Reply:
This is aggregating submissions for export, and so should only `finish` when the submission is committed to the ledger. In other words, no, we need to do this earlier (ideally directly after writing).